### PR TITLE
[INFINITY-2061] Fix target config selection for pod recovery

### PR DESCRIPTION
**Pull request by gabrielhartmann**

### CODE REVIEW

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java [hunk 1]
```diff
@@ -436,4 +438,30 @@ private static void logOutcome(StringBuilder stringBuilder, EvaluationOutcome ou
 
         return null;
     }
+
+    @VisibleForTesting
+    UUID getTargetConfig(PodInstanceRequirement podInstanceRequirement, Collection<Protos.TaskInfo> taskInfos) {
+        if (podInstanceRequirement.getRecoveryType().equals(RecoveryType.NONE) || taskInfos.isEmpty()) {
```
Comment (elezar) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java]: Just for my understanding:
- This also handles the initial case where a pod is first launched?
- The `taskInfos` passed are the tasks defined in the spec, and the `recoveryType` is explicitly set where required (defaulting to `NONE`)?

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java [hunk 2]
```diff
@@ -436,4 +438,30 @@ private static void logOutcome(StringBuilder stringBuilder, EvaluationOutcome ou
...
+            // representative of the whole of the pod. If tasks in the same pod with a goal
+            // state of RUNNING had different target configurations this should be rectified
+            // in any case, so it is doubly proper to choose a single target configuration as
+            // representative of the whole pod's target configuration.
+
+            Protos.TaskInfo taskInfo = taskInfos.stream().findFirst().get();
+            try {
+                return new TaskLabelReader(taskInfo).getTargetConfiguration();
+            } catch (TaskException e) {
+                logger.error(
```
Comment (elezar) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java]: Should we add information on the exception here too?

Comment (nickbp) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java]: May make sense to log the `targetConfigId` as well

File: sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java (hunk 1 above)
Comment (gabrielhartmann) [inline sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/OfferEvaluator.java]: Correct.  PodInstanceRequirements [default to NONE as recoveryType](https://github.com/mesosphere/dcos-commons/blob/master/sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/PodInstanceRequirement.java#L98).  When type is `NONE`, that's the initial launch condition.

**Pull request merged**